frontends/concrete-python/examples/tfhers-cml/test.py

```diff
@@ -0,0 +1,191 @@
+LWE_NOISE_DISTR = 0
+GLWE_NOISE_DISTR = 2.168404344971009e-19
+
+assert GLWE_DIM == 1, "glwe dim must be 1"
+
+# Options: the user can change the following
+# FIXME: explain FHEUINT_PRECISION, ie can it be changed
+FHEUINT_PRECISION = 8
+tfhers_params, tfhers_type, tfhers_int = get_tfhers_params_and_type_and_int(FHEUINT_PRECISION)
+
+# Describe the function you want to apply, on Concrete ciphertexts
+def server_side_function_in_concrete(concrete_x, concrete_y):
+    return (concrete_x + concrete_y) % 47
+
+# The user must specify the range of the TFHE-rs inputs
+# FIXME: why can't we set the limit at 256? It's needed for FHEUint8
+inputset_of_tfhe_rs_inputs = [(tfhers_int(randint(128)),
+                               tfhers_int(randint(128))) for _ in range(100)]
+
+# End of options
+
+# This is the compiled function: user doesn't have to change this, except to
+# add more inputs (ie, tfhers_z etc)
+def function_to_run_in_concrete(tfhers_x, tfhers_y):
+
+    # Here, tfhers_x and tfhers_y are in TFHE-rs format
+
+    concrete_x = tfhers.to_native(tfhers_x)
+    concrete_y = tfhers.to_native(tfhers_y)
+
+    # Here, concrete_x and concrete_y are in Concrete format
+
+    # Here we can apply whatever function we want in Concrete
+    concrete_res = server_side_function_in_concrete(concrete_x, concrete_y)
+
+    # Here, concrete_res is in Concrete format
+
+    tfhers_res = tfhers.from_native(
+        concrete_res, tfhers_type
+    )  # we have to specify the type we want to convert to
+
+
+    # Here, tfhers_res is in TFHE-rs format
+
+    return tfhers_res
+
+# This is where we compile the function with Concrete: user doesn't have to
+# change this, except to add more inputs (ie, tfhers_z etc)
+def compile_concrete_function():
+    compiler = fhe.Compiler(function_to_run_in_concrete, {"tfhers_x": "encrypted", "tfhers_y": "encrypted"})
+
+    circuit = compiler.compile(inputset_of_tfhe_rs_inputs)
+
+    tfhers_bridge = tfhers.new_bridge(circuit=circuit)
+    return circuit, tfhers_bridge
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option("-s", "--secret-key", type=str, required=True)
+@click.option("-k", "--concrete-keyset-path", type=str, required=True)
+# This is where we generate the evaluation key at the Concrete format, from the
+# secret key coming from TFHE-rs, on the client side
+def keygen(secret_key: str, concrete_keyset_path: str):
+    """Concrete Key Generation"""
+
+    # Compile the Concrete function
+    circuit, tfhers_bridge = compile_concrete_function()
+
+    if os.path.exists(concrete_keyset_path):
+        os.remove(concrete_keyset_path)
+
+    # Load the initial secret key to use for keygen
+    with open(
+        secret_key,
+        "rb",
+    ) as f:
+        buff = f.read()
+
+    input_idx_to_key = {0: buff, 1: buff}
+    tfhers_bridge.keygen_with_initial_keys(input_idx_to_key_buffer=input_idx_to_key)
+
+    # FIXME: remove the secret key before saving. The secret key can be used for
```

Comment:
I think there is an issue here, we shouldn't save secret key, it's dangerous if it goes into production, I'll fill an issue for the examples/tfhers file

Reply:
ie, https://github.com/zama-ai/concrete-internal/issues/874